scalafix/rules/src/main/scala/rsc/rules/RscCompat.scala

```diff
@@ -11,39 +11,39 @@ import rsc.rules.syntax._
 import scala.meta._
 import scala.meta.contrib._
 import scala.meta.internal.{semanticdb => s}
-import scalafix.internal.util._
-import scalafix.internal.v0._
-import scalafix.rule._
-import scalafix.syntax._
+import scalafix.internal.v1._
+import scalafix.lint.Diagnostic
+import scalafix.patch.Patch
+import scalafix.v1.{Configuration, Rule}
 import scalafix.util.TokenOps
-import scalafix.v0._
+import scalafix.v1.SemanticDocument
+import scalafix.v1._
 
-case class RscCompat(legacyIndex: SemanticdbIndex, config: RscCompatConfig)
-    extends SemanticdbRule(legacyIndex, "RscCompat") {
-  def this(legacyIndex: SemanticdbIndex) = {
-    this(legacyIndex, RscCompatConfig.default)
-  }
+case class RscCompat(config: RscCompatConfig)
+    extends SemanticdbRule( "RscCompat") {
+  def this() = this(RscCompatConfig.default)
 
-  override def init(config: Conf): Configured[Rule] = {
-    config
+  override def withConfiguration(config: Configuration): Configured[Rule] = {
+    config.conf
       .getOrElse("rscCompat", "RscCompat")(RscCompatConfig.default)
-      .map(RscCompat(legacyIndex, _))
+      .map(RscCompat(_))
   }
 
-  override def fix(ctx: RuleCtx): Patch = {
-    val targets = collectRewriteTargets(ctx)
-    targets.map(ascribeReturnType(ctx, _)).asPatch
+  override def fix(implicit doc: SemanticDocument): Patch = {
+    val targets = collectRewriteTargets(doc)
+    targets.map(ascribeReturnType(doc, _)).asPatch
   }
 
-  private case class RewriteTarget(
+  case class RewriteTarget(
       env: Env,
       before: Token,
       name: Name,
       after: Token,
       body: Term,
       parens: Boolean)
 
-  private def collectRewriteTargets(ctx: RuleCtx): List[RewriteTarget] = {
+  def collectRewriteTargets(doc: SemanticDocument): List[RewriteTarget] = {
+    implicit val impDoc = doc
```

Comment:
Looks like the `doc` parameter should be implicit after the migration to v1. A dummy implicit val looks non-idiomatic.